### PR TITLE
Fix Tests when no config is specified in config.json

### PR DESCRIPTION
**Pull request by noesberger**
When no config is specified, the conversion of [int], [string] or [bool] is generating an error. I've now putted in all the tests a check for the $desired. When nothing is specified in the config, the $desired is also not specified and with -verbose the following is shown as expected:
VERBOSE: Due to null config value, skipping test

I'm also removed the CmdletBinding(SupportsShouldProcess) because it's not used in the code and visual studio code is reporting this as an issue.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,4 +39,8 @@ Tests/TestVars
 .Spotlight-V100
 .Trashes
 ehthumbs.db
-Thumbs.db
\ No newline at end of file
+Thumbs.db
+
+# Custom Test Config #
+######################
+Vester/Configs/config.json
```
Comment (brianbunke) [inline .gitignore]: We need this. ❤️

File: Vester/Public/Invoke-Vester.ps1 [hunk 2]
```diff
@@ -70,8 +70,7 @@ function Invoke-Vester {
     .LINK
     https://github.com/WahlNetwork/Vester
     #>
-    [CmdletBinding(SupportsShouldProcess = $true,
-                   ConfirmImpact = 'Medium')]
+
```
Comment (brianbunke) [inline Vester/Public/Invoke-Vester.ps1]: This is not unused. It passes `-WhatIf` support through to `VesterTemplate.Tests.ps1`

File: Vester/Tests/Cluster/HA-Enabled.Vester.ps1 [hunk 3]
```diff
@@ -5,7 +5,9 @@
 $Title = 'HA State'
 
 # The config entry stating the desired values
-[bool]$Desired = $cfg.cluster.haenable
+if($cfg.cluster.haenable){
```
Comment (brianbunke) [inline Vester/Tests/Cluster/HA-Enabled.Vester.ps1]: This evaluates `$null` = `$false` = `0`, and the last two are valid values. More on the test changes in general review comments.

File: .gitignore (hunk 1 above)
Comment (noesberger) [inline .gitignore]: Yes, the config.json is needed when you run tests. But why it should be in the git-repo? Here is only user specific data in the config.

**Pull request closed without merging**